core/00_infra.js

```diff
@@ -166,27 +140,17 @@
         throw "Missing promise in map @ " + promiseId;
       }
       MapPrototypeDelete(promiseMap, promiseId);
-      return promise;
-    }
-    // Otherwise take from ring
-    const idx = promiseId % RING_SIZE;
-    const promise = promiseRing[idx];
-    if (!promise) {
-      throw "Missing promise in ring @ " + promiseId;
+      promise(res);
+    } else {
+      // Otherwise take from ring
+      const idx = promiseId % RING_SIZE;
+      const promise = promiseRing[idx];
+      if (!promise) {
+        throw "Missing promise in ring @ " + promiseId;
+      }
+      promiseRing[idx] = NO_PROMISE;
+      promise(res);
     }
-    promiseRing[idx] = NO_PROMISE;
-    return promise;
-  }
-
-  function newPromise() {
```

Comment:
Why is this method gone?

Reply:
It got absorbed into the `setPromise` method to avoid method call overhead.

Reply:
We skip the intermediate value for the promise resolver and just place the resolve method directly into the promise map now:

https://github.com/denoland/deno_core/pull/573/files#diff-c65a34b915562b309be0123993a975d6f83222c66390db083f403409d9251430R123

Reply:
Huh, did we never use `reject` method?

Side note: we should bench and see if using `Promise.withResolvers` is faster than doing it manually

Reply:
I did that a while back and it ended up being slower 😢